### ISSUE TITLE
Magic comments are not highlighted properly

### ISSUE THREAD

**Issue by schtandard** [texstudio-org/texstudio#2267]
### Environment
- TeXstudio: 4.2.3
- Qt: 6.3.0
- OS: Windows 11
- TeX distribution: Tex Live 2022

### Description
A magic comment like
```latex
% !TeX program = lualatex
```
should be highlighted using the `magicComment` styling but instead is just highlighted using `comment`.
I believe this was introduced when the comment highlighting was moved to the syntax checker in 0eb45d8.

### How to reproduce
Post these two lines in a `tex` file. The two lines are styled the same but should not.
```latex
% !TeX program = lualatex     <- magic comment
% foo bar baz                 <- normal comment
```

**Comment by octaeder** [comment]
can confirme for 4.2.3 Win10. If I increase prio to 0 in the config here it works
![grafik](https://user-images.githubusercontent.com/102688820/164673673-3e9c225a-c753-48df-a4db-abfad8a25e6a.png)
But colors are not very different
![grafik](https://user-images.githubusercontent.com/102688820/164674356-62e93996-0bbe-4117-9e54-9eaac3e4d297.png)

**Comment by sunderme** [comment]
fixed

**Comment by octaeder** [comment]
@schtandard please set prio to 0 as shown above for magicComment and also in the next line for commentTodo.

**Comment by schtandard** [comment]
@octaeder That works.

@sunderme Did you set the priority in the dark theme to `5` instead of `0` (as in the light theme) on purpose?

**Comment by octaeder** [comment]
yes, he did so. You can see it here [1c6bc35](https://github.com/texstudio-org/texstudio/commit/1c6bc356518f75eb852ceea5b0810e8bf79d49dd) (whatch out for this number with the green checkmark above on the right of the comment by sunderme at 17:52 yesterday). Changes are:
```
  <format id="magicComment" >
    <bold>false</bold>
    <italic>false</italic>
    <overline>false</overline>
    <underline>false</underline>
    <strikeout>false</strikeout>
    <waveUnderline>false</waveUnderline>
    <foreground>#59a56c</foreground>
    <priority>5</priority>                   new
  </format>
  <format id="commentTodo" >
    <bold>true</bold>
    <italic>false</italic>
    <overline>false</overline>
    <underline>false</underline>
    <strikeout>false</strikeout>
    <waveUnderline>false</waveUnderline>
    <foreground>#a55992</foreground>
    <priority>5</priority>                   new
  </format>
```
Same two changes as I mentioned above, one for magicComment and one for commentTodo.

**Comment by schtandard** [comment]
@octaeder You misunderstand. I get that he made these changes, I was just surprised that the priority is set to `0` for the light theme but to `5` for the dark theme. I would have expected `0` in both cases. Maybe there's a reason, I was just asking to double-check.

**Comment by octaeder** [comment]
Oh sorry, maybe sunderme can tell.

**Comment by octaeder** [comment]
Well observed, @schtandard, i's corrected by sunderme here [348e51f](https://github.com/texstudio-org/texstudio/commit/348e51fe618dc32eadd56f74bc40d5b3d45a1beb):
```
  <format id="magicComment" >
    <bold>false</bold>
    <italic>false</italic>
    <overline>false</overline>
    <underline>false</underline>
    <strikeout>false</strikeout>
    <waveUnderline>false</waveUnderline>
    <foreground>#59a56c</foreground>
    <priority>0</priority>
  </format>
  <format id="commentTodo" >
    <bold>true</bold>
    <italic>false</italic>
    <overline>false</overline>
    <underline>false</underline>
    <strikeout>false</strikeout>
    <waveUnderline>false</waveUnderline>
    <foreground>#a55992</foreground>
    <priority>0</priority>
  </format>
